.travis.yml

```diff
@@ -16,6 +16,9 @@
 language: java
 
 sudo: false
+cache:
+  directories:
+    - .spark-dist
```

Comment:
Yes, as I mentioned above - that would be the next thing to do. But I would prefer doing it in separate PR. 

The way it works (tar everything to s3 at the beginning\end of the buid) - I'm not sure it will work for our size of `.m2`, so some experiments will be needed.

And cache is alrady polluted by prev experiments so I was waiting for [INFRA-11580](https://issues.apache.org/jira/browse/INFRA-11580) to reset it, before moving to m2.